src/main/scala/is/hail/utils/SimpleTabulationHash32.scala

```diff
@@ -0,0 +1,131 @@
+package is.hail.utils
+
+import java.security.SecureRandom
+
+object UnivHash32 {
+  def apply(outBits: Int = 32): UnivHash32 = {
```

Comment:
I'm apprehensive about using an unseeded source of randomness. Would methods use this function? I suspect we at least want to test high-level statistical method with a particular seed.

Reply:
Right, this is something I spent some time thinking about.

All hash functions (at least of the kind I'm considering right now) take seeds. For `UnivHash32` functions, for example, you can create them directly with an explicit seed, or use the companion object to pick a random seed for you. My thought was that we would use explicit seeds for tests, and fully random ones in practice.

It gets a bit more complicated with the more powerful hash functions. For example, the seed of `SimpleTabulationHash32` is a 4KB table. If that table isn't sufficiently random, then you lose the guarantees of the behavior of the hash function. `SecureRandom` is drawing from the kernel's entropy pool, so it is definitely random enough. For testing, it might not matter much that the seed table for a tabulation hash isn't random enough for the theoretical guarantees, so I could add another constructor that fills the table with MersenneTwister for use in tests.

There are pseudorandom number generators strong enough to fill a seed table from a much smaller seed, and implementing at least one of them is on my list. Once we have those, it will be easier to deterministically seed a tabulation hash function.

Reply:
Assuming we could generate a satisfactory table, is there any reason to not always seed our randomness?

There are sources of randomness outside our control, such as the network, but I feel like, as much as we can, we should try to produce a deterministic system.

Reply:
This is a question I'd very much like to discuss more with everybody. I don't know the answer. I can see arguments on both sides.

 I'd like to lead some more Hail lab meetings on randomized algorithms, which might be a good way to start this discussion.

I did come across an amusing quote on the subject:
"Anyone attempting to generate random numbers by deterministic means is, of course, living in a state of sin." -- John Von Neumann

Reply:
I think more lab meetings is a very productive way to proceed. I don't really have the knowledge necessary to argue this point. I only have an intuition that non-determinism is the enemy of the engineer.